### PR TITLE
Fixing - IDENTITY-3156

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
**Fixing - IDENTITY-3156**

Added the "GrantType" column to the IDN_OAUTH2_ACCESS_TOKEN.

https://wso2.org/jira/browse/IDENTITY-3156

### CODE REVIEW

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 1]
```diff
@@ -38,23 +39,8 @@
...
-import java.util.Calendar;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.TimeZone;
-import java.util.UUID;
+import java.sql.*;
+import java.util.*;
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: You haven't installed the wso2 code-style template. It is mentioned in the doc I have shared with you the other day.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql [hunk 2]
```diff
@@ -217,7 +218,7 @@ CREATE TABLE IDN_THRIFT_SESSION (
 CREATE TABLE IDN_AUTH_SESSION_STORE (
            SESSION_ID VARCHAR (100) NOT NULL,
            SESSION_TYPE VARCHAR(100) NOT NULL,
-	   OPERATION VARCHAR(10) NOT NULL,
+	         OPERATION VARCHAR(10) NOT NULL,
```
Comment (cdwijayarathna) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql]: not formatted

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/h2.sql [hunk 3]
```diff
@@ -97,7 +98,7 @@ CREATE TABLE IF NOT EXISTS IDN_OAUTH2_SCOPE (
             NAME VARCHAR(255) NULL,
             DESCRIPTION VARCHAR(512) NULL,
             TENANT_ID INTEGER NOT NULL DEFAULT 0,
-	    ROLES VARCHAR (500) NULL,
```
Comment (cdwijayarathna) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/h2.sql]: formatting issues

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 4]
```diff
@@ -228,89 +230,92 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
-                    revokeRespDTO.setError(true);
-                    revokeRespDTO.setErrorCode(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT);
-                    revokeRespDTO.setErrorMsg("Unauthorized Client");
-                    return revokeRespDTO;
-                }
+
                 boolean refreshTokenFirst = false;
-                if (revokeRequestDTO.getToken_type() != null && REFRESH_TOKEN.equals(revokeRequestDTO.getToken_type())) {
+                if (revokeRequestDTO.getToken_type() != null &&
+                        REFRESH_TOKEN.equals(revokeRequestDTO.getToken_type())) {
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: You can use StringUtils.equals and avoid null check

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 5]
```diff
@@ -228,89 +230,92 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
+
+                String grantType = StringUtils.EMPTY;
+
+                if (accessTokenDO != null) {
+                    grantType = accessTokenDO.getGrantType();
+                } else if (refreshTokenDO != null) {
+                    grantType = refreshTokenDO.getGrantType();
+                }
+
+                if (!IMPLICIT.equals(grantType) &&
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Use StringUtils.equals and avoid null check

**Comment by johannnallathamby** [comment]
Length 25 for GrantType column may not be enough. Shall we increase it to 50

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 6]
```diff
@@ -228,89 +230,92 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
-                            org.wso2.carbon.identity.oauth.OAuthUtil.clearOAuthCache(
-                                    revokeRequestDTO.getConsumerKey(), accessTokenDO.getAuthzUser());
-                            org.wso2.carbon.identity.oauth.OAuthUtil.clearOAuthCache(revokeRequestDTO.getToken());
-                            tokenMgtDAO.revokeTokens(new String[]{revokeRequestDTO.getToken()});
-                            addRevokeResponseHeaders(revokeResponseDTO, accessTokenDO.getAccessToken(),
-                                    accessTokenDO.getRefreshToken(), refreshTokenDO.getAuthorizedUser());
-                        }
+                    refreshTokenDO = tokenMgtDAO
+                            .validateRefreshToken(revokeRequestDTO.getConsumerKey(), revokeRequestDTO.getToken());
+                    if (refreshTokenDO == null) {
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Seems checking for null only isn't enough. In lines 251 to 255 of the original code we are doing much more conditional checks. Specifically we are not only checking if a token exists but we are also checking its status.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 7]
```diff
@@ -228,89 +230,92 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
-                                    OAuth2Util.buildScopeString(refreshTokenDO.getScope()));
-                            org.wso2.carbon.identity.oauth.OAuthUtil.clearOAuthCache(
-                                    revokeRequestDTO.getConsumerKey(), refreshTokenDO.getAuthorizedUser());
-                            org.wso2.carbon.identity.oauth.OAuthUtil.clearOAuthCache(refreshTokenDO.getAccessToken());
-                            tokenMgtDAO.revokeTokens(new String[]{refreshTokenDO.getAccessToken()});
-                            addRevokeResponseHeaders(revokeResponseDTO, refreshTokenDO.getAccessToken(),
-                                    revokeRequestDTO.getToken(), refreshTokenDO.getAuthorizedUser());
-                        }
+                    accessTokenDO = tokenMgtDAO.retrieveAccessToken(revokeRequestDTO.getToken(), true);
+                    if (accessTokenDO == null) {
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Same comment above applies here as well.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 8]
```diff
@@ -62,7 +62,9 @@
 @SuppressWarnings("unused")
 public class OAuth2Service extends AbstractAdmin {
 
+    public static final String IMPLICIT = "implicit";
     public static final String REFRESH_TOKEN = "refresh_token";
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Move both of these constants to OAuthConstants in oauth.common bundle

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java [hunk 9]
```diff
@@ -43,23 +43,35 @@
 
     private static Log log = LogFactory.getLog(TokenResponseTypeHandler.class);
 
+    public static final String TOKEN = "token";
+    public static final String IMPLICIT = "implicit";
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Can remove this constant if u have moved it to oauth.common

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 10]
```diff
@@ -682,23 +694,34 @@ public RefreshTokenValidationDataDO validateRefreshToken(String consumerKey, Str
...
             }
+
             if (scopes.size() > 0 && validationDataDO != null){
                 validationDataDO.setScope((String[])ArrayUtils.addAll(validationDataDO.getScope(),  
                     scopes.toArray(new String[scopes.size()])));
             }
+
             connection.commit();
+
+        } catch (IdentityException e) {
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: IdentityException should not be throw here. Please check.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 11]
```diff
@@ -793,25 +824,37 @@ public AccessTokenDO retrieveAccessToken(String accessTokenIdentifier, boolean i
...
             }
+
             if (scopes.size() > 0 && dataDO != null){
                 dataDO.setScope((String[])ArrayUtils.addAll(dataDO.getScope(), 
                      scopes.toArray(new String[scopes.size()])));
             }
+
             connection.commit();
+
+        } catch (IdentityException e) {
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: IdentityException should not be throw here. Please check. IdentityDatabaseUtil.getDBConnection() now throws a IdentityRuntimeException which you don't need to handle

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 12]
```diff
@@ -762,13 +786,19 @@ public AccessTokenDO retrieveAccessToken(String accessTokenIdentifier, boolean i
...
             }
-            String sql = SQLQueries.RETRIEVE_ACTIVE_ACCESS_TOKEN;
+
+            String sql;
+
             if (includeExpired) {
                 sql = SQLQueries.RETRIEVE_ACTIVE_EXPIRED_ACCESS_TOKEN;
+            } else {
+                sql = SQLQueries.RETRIEVE_ACTIVE_ACCESS_TOKEN;
             }
```
Comment (johannnallathamby) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Why is this change needed ? Earlier it was initialized at declaration and changed under if condition. Now it is initialized under if or else. Is this change needed in some way. If not remove it.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 13]
```diff
@@ -216,10 +219,13 @@ public void storeAccessToken(String accessToken, String consumerKey,
             prepStmt.setString(11, accessTokenDO.getTokenState());
             prepStmt.setString(12, accessTokenDO.getTokenType());
             prepStmt.setString(13, accessTokenDO.getTokenId());
-            prepStmt.setString(14, persistenceProcessor.getProcessedClientId(consumerKey));
+            prepStmt.setString(14, accessTokenDO.getGrantType());
+            prepStmt.setString(15, persistenceProcessor.getProcessedClientId(consumerKey));
```
Comment (madurangasiriwardena) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: TOKEN_ID is 14th parameter and GRANT_TYPE is 15th parameter. check with the sql query

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 12 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: This is due to the strange behavior of the string replacement. I don't know whether it is a isolated incident in my PC or a JVM issue.

**Comment by JKAUSHALYA** [comment]
Change the Database column size of the GrantType to 50.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 14]
```diff
@@ -228,89 +228,110 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
-                            tokenMgtDAO.revokeTokens(new String[]{revokeRequestDTO.getToken()});
-                            addRevokeResponseHeaders(revokeResponseDTO, accessTokenDO.getAccessToken(),
-                                    accessTokenDO.getRefreshToken(), refreshTokenDO.getAuthorizedUser());
-                        }
+
+                    refreshTokenDO = tokenMgtDAO
+                            .validateRefreshToken(revokeRequestDTO.getConsumerKey(), revokeRequestDTO.getToken());
+
+                    if (refreshTokenDO == null ||
+                            StringUtils.isEmpty(refreshTokenDO.getRefreshTokenState()) ||
```
Comment (hasinthaindrajee) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Format code

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 15]
```diff
@@ -228,89 +228,110 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
-                            addRevokeResponseHeaders(revokeResponseDTO, refreshTokenDO.getAccessToken(),
-                                    revokeRequestDTO.getToken(), refreshTokenDO.getAuthorizedUser());
+                    accessTokenDO = tokenMgtDAO.retrieveAccessToken(revokeRequestDTO.getToken(), true);
+                    if (accessTokenDO == null) {
+
+                        refreshTokenDO = tokenMgtDAO
+                                .validateRefreshToken(revokeRequestDTO.getConsumerKey(), revokeRequestDTO.getToken());
+
+                        if (refreshTokenDO == null ||
+                                StringUtils.isEmpty(refreshTokenDO.getRefreshTokenState()) ||
```
Comment (hasinthaindrajee) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Format Code

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java [hunk 16]
```diff
@@ -49,17 +50,27 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
...
+
         respDTO.setCallbackURI(authorizationReqDTO.getCallbackUrl());
+
         String consumerKey = authorizationReqDTO.getConsumerKey();
         String authorizedUser = authorizationReqDTO.getUsername();
         String oAuthCacheKeyString;
+
+        String responseType = oauthAuthzMsgCtx.getAuthorizationReqDTO().getResponseType();
+
+        String grantType = StringUtils.equals(OAuthConstants.GrantTypes.TOKEN, responseType) ?
```
Comment (hasinthaindrajee) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Use if then else to improve readablity

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 17]
```diff
@@ -333,6 +333,8 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
                         .getRefreshTokenValidityPeriodInSeconds() * 1000;
             }
 
+            String grantType = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getGrantType();
```
Comment (hasinthaindrajee) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Is there a possibility to tokReqMsgCtx.getOauth2AccessTokenReqDTO() be null

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 18]
```diff
@@ -342,10 +344,11 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
             newAccessTokenDO.setTokenState(OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE);
             newAccessTokenDO.setTenantID(tokReqMsgCtx.getTenantID());
             newAccessTokenDO.setTokenId(UUID.randomUUID().toString());
+            newAccessTokenDO.setGrantType(grantType);
 
             // Persist the access token in database
             storeAccessToken(oAuth2AccessTokenReqDTO, userStoreDomain, newAccessTokenDO, newAccessToken,
-                             existingAccessTokenDO);
+                    existingAccessTokenDO);
```
Comment (hasinthaindrajee) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Format code

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java [hunk 19]
```diff
@@ -168,21 +179,29 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
                                     " was added to cache for cache key : " + cacheKey.getCacheKeyString());
                         }
                     }
+
                     respDTO.setAccessToken(existingAccessTokenDO.getAccessToken());
+
                     if(expiryTime > 0){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 20]
```diff
@@ -679,23 +691,30 @@ public RefreshTokenValidationDataDO validateRefreshToken(String consumerKey, Str
...
+                    validationDataDO.setGrantType(resultSet.getString(10));
+
                 } else {
                     scopes.add(resultSet.getString(5));
                 }
+
                 iterateId++;
             }
+
             if (scopes.size() > 0 && validationDataDO != null){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 21]
```diff
@@ -790,25 +817,34 @@ public AccessTokenDO retrieveAccessToken(String accessTokenIdentifier, boolean i
...
+                    dataDO.setGrantType(grantType);
+
+                } else {
                     scopes.add(resultSet.getString(5));
                 }
+
                 iterateId++;
             }
+
             if (scopes.size() > 0 && dataDO != null){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java [hunk 22]
```diff
@@ -790,25 +817,34 @@ public AccessTokenDO retrieveAccessToken(String accessTokenIdentifier, boolean i
...
                 dataDO.setScope((String[])ArrayUtils.addAll(dataDO.getScope(), 
                      scopes.toArray(new String[scopes.size()])));
             }
+
             connection.commit();
+
         } catch (SQLException e) {
             throw new IdentityOAuth2Exception("Error when retrieving Access Token : " + accessTokenIdentifier, e);
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Remove access token info from the message.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 23]
```diff
@@ -342,10 +344,11 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
...
 
             if (log.isDebugEnabled()) {
                 log.debug("Persisted Access Token : " + newAccessToken + " for " +
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Remove the token info from the log.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 24]
```diff
@@ -228,89 +228,110 @@ public OAuth2AccessTokenRespDTO issueAccessToken(OAuth2AccessTokenReqDTO tokenRe
...
+                    revokeRespDTO.setError(true);
+                    revokeRespDTO.setErrorCode(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT);
+                    revokeRespDTO.setErrorMsg("Unauthorized Client");
+
+                    return revokeRespDTO;
+                }
+
+                if (refreshTokenDO != null) {
+
+                    org.wso2.carbon.identity.oauth.OAuthUtil
```
Comment (malithie) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: import the OAuthUtil class at imports. That will improve the readability of the code

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java (hunk 15 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java (hunk 16 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java (hunk 17 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java (hunk 18 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java (hunk 19 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 20 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 21 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java (hunk 22 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java]: Fixed in #1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java (hunk 23 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: https://github.com/wso2/carbon-identity/pull/1486

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java (hunk 24 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java]: Fixed in #1690
